src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -54,15 +54,60 @@ static ucs_status_t uct_cuda_ipc_rkey_unpack(uct_md_component_t *mdc,
                                              void **handle_p)
 {
     uct_cuda_ipc_key_t *packed = (uct_cuda_ipc_key_t *) rkey_buffer;
+    uct_cuda_ipc_md_t *md = mdc->priv;
     uct_cuda_ipc_key_t *key;
     ucs_status_t status;
     CUdevice cu_device;
+    CUdevice own_cu_device;
+    CUdevice peer_cu_device;
+    CUuuid own_uuid;
     int peer_accessble;
+    int device_count;
+    int i;
 
     UCT_CUDA_IPC_GET_DEVICE(cu_device);
+    
+    if (NULL == md->uuid_map) {
```

Comment:
why can't you setup this map in md_open itself?

Reply:
Device may not be selected even though `CUDA_VISIBLE_DEVICES` are specified, right? This would be the case when device selection happens after `ucp_init` but `UCX_MEMTYPE_CACHE` is turned off.

Reply:
this table is not specific to selected device right? here we are building a table of uuid for all the devices.

Reply:
You're right. I'll push this to `md_open`.